File: xinyu-web/src/api/modules/stats.ts
import { get } from '@/api/request'
import type { UsageStatsVO } from '@/types/api'

/**
 * 用量统计接口（M2 Token 统计, 需登录; 仅返回当前用户本人数据）
 */
export const statsApi = {
  /** 今日 + 累计 调用次数 / token / 估算成本 */
  usage(): Promise<UsageStatsVO> {
    return get('/stats/usage')
  },
}

File: xinyu-backend/app/api/stats.py
"""用量统计接口 — 对应 Java StatsController"""

from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.api.deps import get_current_user_id
from app.core.database import get_db
from app.schemas.common import Result
from app.schemas.stats import UsageStatsVO
from app.services import stats_service

router = APIRouter(prefix="/api/stats", tags=["stats"])


@router.get("/usage")
async def usage(
    user_id: int = Depends(get_current_user_id), db: AsyncSession = Depends(get_db)
) -> Result[UsageStatsVO]:
    """当前用户用量统计: 今日 + 累计 调用次数 / token / 估算成本"""
    return Result.ok(await stats_service.get_usage(db, user_id))
